### DISCUSSION TITLE
ReactiveObject vs. AvaloniaObject

### DISCUSSION THREAD

**Discussion by najak3d** [AvaloniaUI/Avalonia#7704] [Q&A] [answered]
The API for binding/observables for AvaloniaObject seem clear.  Code like this works on Avalonia Objects:

```c#
	var binding = new AD.Binding { Source = _dockBase, Path = "ActiveDockable.FocusedDockable", FallbackValue = "" };
	txtBlk.Bind(TextBlock.TextProperty, binding);

	mainDock.Bind(DAC.DockControl.LayoutProperty, new AD.Binding() { Source = _vm, Path = "Layout" });
	navBtn.Bind(Button.CommandParameterProperty, navId.GetObservable<string>(TextBox.TextProperty));
```

**But this code doesn't work with ReactiveObjects.    What are the equivalent C# methods for Binding with ReactiveUI Objects/Properties?**   

We are working with pure C#, not much AXAML (that's the goal).

Also, **for Avalonia namespace, is there an equivalent of "ReactiveUI.ReactiveCommand.Create(Action action)"  ??**

We are working on demo code to release to Github, which shows the pure-C# usage of Avalonia, and want to show the differences in C# between using AvaloniaObjects vs. ReactiveUI.

Is there any link online that compares these two approaches side-by-side?

**Comment by jp2masa** [comment, marked as answer]
`AvaloniaObject`s are objects of the UI framework, while `ReactiveObject`s are view models, i.e. do not depend on a specific UI framework.

However, ReactiveUI has some integration with UI frameworks, and it's possible to bind from code-behind, see [ReactiveUI data binding docs](https://www.reactiveui.net/docs/handbook/data-binding/), including [the part specific to Avalonia](https://www.reactiveui.net/docs/handbook/data-binding/avalonia).

> Also, for Avalonia namespace, is there an equivalent of "ReactiveUI.ReactiveCommand.Create(Action action)" ??

Avalonia consumes `ICommand` implementations, and `ReactiveCommand` is one that works for ReactiveUI, but if you need it you can create your custom implementation, like [the one in MiniMvvm created for Avalonia samples](https://github.com/AvaloniaUI/Avalonia/blob/237a9a41ea919693b66c425c455148d322385df9/samples/MiniMvvm/MiniCommand.cs).
